docs/en/migrations/bigquery.md

```diff
@@ -0,0 +1,110 @@
+        format = 'PARQUET',
+        overwrite = true
+    )
+AS (
+    SELECT * FROM mytable
+);
+```
+
+In the above query, we export our BigQuery table to the [Parquet data format](https://parquet.apache.org/). We also have a `*` character in our `uri` parameter. This ensures the output is sharded into multiple files, with a numerically increasing suffix, should the export exceed 1GB of data.
+
+This approach has a number of advantages:
+
+- Google allows up to 50TB per day to be exported to GCS for free. Users only pay for GCS storage.
+- Exports produce multiple files automatically, limiting each to a maximum of 1GB of table data. This is beneficial to ClickHouse since it allows imports to be parallelized.
+- Parquet, as a column-oriented format, represents a better interchange format since it is inherently compressed and faster for BigQuery to export and ClickHouse to query
+
+## 2. Importing data into ClickHouse from GCS
+
+Once the export is complete, we can import this data into a ClickHouse table. You can use the [ClickHouse SQL console](/docs/en/integrations/sql-clients/sql-console) or [`clickhouse-client`](/docs/en/integrations/sql-clients/cli) to execute the commands below.
+
+You must first [create your table](/docs/en/sql-reference/statements/create/table) in ClickHouse:
+
+```sql
+CREATE TABLE default.mytable
+(
+	`timestamp` DateTime64(6),
+	`some_text` String
+)
+ENGINE = MergeTree
+ORDER BY (timestamp);
+```
+
+After creating the table, we enable parallel inserts on selects to speed up our export:
+
+```sql
+SET parallel_distributed_insert_select = 1;
+```
+
+Finally, we can insert the data from GCS into our ClickHouse table using the [`INSERT INTO SELECT` command](/docs/en/sql-reference/statements/insert-into#inserting-the-results-of-select), which inserts data into a table based on the results from a `SELECT` query.
+
+To retrieve the data to `INSERT`, we can use the [s3Cluster function](/docs/en/sql-reference/table-functions/s3Cluster) to retrieve data from our GCS bucket since GCS is interoperable with [Amazon S3](https://aws.amazon.com/s3/):
+
+```sql
+INSERT INTO mytable
+SELECT
+    timestamp,
+    ifNull(some_text, '') as some_text
+FROM s3Cluster(
+    'default',
+    'https://storage.googleapis.com/mybucket/mytable/*.parquet.gz',
+    '<ACCESS_ID>',
+    '<SECRET>'
+);
+```
+
+The `ACCESS_ID` and `SECRET` used in the above query is your [HMAC key](https://cloud.google.com/storage/docs/authentication/hmackeys) associated with your GCS bucket.
+
+:::note Use `ifNull` when exporting nullable columns
+In the above query, we use the [`ifNull` function](/docs/en/sql-reference/functions/functions-for-nulls#ifnull) with the `some_text` column to insert data into our ClickHouse table with a default value. You can also make your columns in ClickHouse [`Nullable`](/docs/en/sql-reference/data-types/nullable), but this is not recommended as it may affect negatively performance.
+:::
```

Comment:
Might be worth bringing the following to the users attention:

input_format_null_as_default=1 
input_format_parquet_import_nested=1